lib/timers.js

```diff
@@ -126,36 +122,72 @@ const kRefed = Symbol('refed');
 // always be due to timeout at a later time.
 //
 // Less-than constant time operations are thus contained in two places:
-// TimerWrap's backing libuv timers implementation (a performant heap-based
-// queue), and the object map lookup of a specific list by the duration of
-// timers within (or creation of a new list).
-// However, these operations combined have shown to be trivial in comparison to
-// other alternative timers architectures.
+// The PriorityQueue which manages the order of expiring Timeouts and object
+// map lookup of a specific list by the duration of timers within (or creation
+// of a new list). However, these operations combined have shown to be trivial
+// in comparison to other alternative timers architectures.
 
 
 // Object maps containing linked lists of timers, keyed and sorted by their
 // duration in milliseconds.
 //
-// The difference between these two objects is that the former contains timers
-// that will keep the process open if they are the only thing left, while the
-// latter will not.
-//
 // - key = time in milliseconds
 // - value = linked list
-const refedLists = Object.create(null);
-const unrefedLists = Object.create(null);
+const lists = Object.create(null);
+
+// This is a priority queue with a custom sorting function that first compares
+// the expiry times of two lists and if they're the same then compares their
+// individual IDs to determine which list was created first.
```

Comment:
It seems to me if you have two lists of the same duration then something went very wrong?

How would you tell which list to set a timer of 30ms if there were two lists of 30ms?

Reply:
It's different durations, same expiries. One list is for 120s, the other for 60s but started 60s later.

Reply:
Would tracking the expiry be better? You should be able to check `list[0].expiry` I think?

Reply:
Well, expiry is the first point of comparison, right? So we need a secondary criteria which tells us which timer is newer. This also happens to exactly match how this works currently.

Reply:
Oh, for some reason I thought it would be comparing `msecs`. Guess that works for me.